### DISCUSSION TITLE
Coverage and Unit Test

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#267] [Ideas]
So I will start here :-)

I think it could be useful for developers to know about the acutal state of master branch regarding failed tests as a reference, i.e. to check if their current changes result in this failed test. At the moment one has to run unit test twice to compare with master and actual branch.
We could make an issue of it and pin it. 

It would also be interessing to see the actual coverage to know where there are tests missing.

So what do you think?

**Comment by BonneelP** [comment]
Hello Sebastian, 

Great: a very important topic to start the discussions :) 

For that I have a "simple" solution: the master branch should always validate all the tests. I know that it is not currently the case but I think that it can be a good goal looking forward :) With this in consideration we can create a new release at any time if someone opens an issue on a bug. 

To be able to achieve that we need to investigate Travis to automatically run the tests on the latest version of pyleecan. It is something on our TODO list for ages but hopefully we should be able to investigate that in a near future. 

Best regards,
Pierre

**Comment by SebGue** [comment]
Hello Pierre,
I guessed that there is some tool to Automaten that but I haven't investigated.
So as an intermediäre solution I will start a respektive issue to list all failed tests. Maybe we could also kabel Thema failed in pytest to specifically debug them. BR S
